staging/src/k8s.io/apiserver/pkg/server/filters/with_retry_after.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"net/http"
+	"strings"
+)
+
+var (
+	// health probes and metrics scraping are never rejected, we will continue
+	// serving these requests after shutdown delay duration elapses.
+	excludedPathPrefixes = []string{
+		"/readyz",
+		"/livez",
+		"/healthz",
+		"/metrics",
+	}
+)
+
+// excludeFunc returns true if the request should be excluded,
+// otherwise it returns false.
+// excluded requests are always served.
+type excludeFunc func(*http.Request) bool
+
+// retryAfterParams dictates how the Retry-After response is constructed
+type retryAfterParams struct {
+	// TearDownConnection is true when we should send a
+	// 'Connection: close' header in the response.
+	TearDownConnection bool
+
+	// Message describes why Retry-After response has been sent by the server
+	Message string
+}
+
+// retryAfterConditionFunc returns true if the requests should
+// be rejected once certain conditions are met.
+// The retryAfterParams returned contains instructions on how to
+// construct the Retry-After response.
+type retryAfterConditionFunc func() (*retryAfterParams, bool)
+
+func (f retryAfterConditionFunc) ShouldRespondWithRetryAfter() (*retryAfterParams, bool) {
+	return f()
+}
```

Comment:
odd construct. Just use `retryAfterConditionFunc()` directly. No need for a method.

Reply:
done